src/compiler/tracing.ts

```diff
@@ -140,12 +140,15 @@ namespace ts.tracingEnabled { // eslint-disable-line one-namespace-per-file
         }
         eventStack.length = 0;
     }
+    // sample every 10ms
+    const sample = 1000 * 10;
     function writeStackEvent(index: number, endTime: number) {
         const { phase, name, args, time, separateBeginAndEnd } = eventStack[index];
         if (separateBeginAndEnd) {
             writeEvent("E", phase, name, args, /*extras*/ undefined, endTime);
         }
-        else {
+        // test if [ts,ts+dur] straddles a sampling point

```

Comment:
Is there an important difference between `[ts,ts+dur]` and `[ts,ts+dur)`?  Also, `[time,endTime]` might be clearer in context.

Reply:
Technically it is `[...)`, but they're floating point numbers, so that doesn't make any difference; but I changed it anyway...